tests/extmod/vfs_posix.py

```diff
@@ -130,6 +130,70 @@ def write_files_without_closing():
 except OSError:
     print("remove OSError")
 
+# construct new VfsPosix with absolute path argument
```

Comment:
This commit "Add more tests for relative paths, currently failing." adds a lot to the existing test file and makes it too complex.  So please can you put these new tests in a separate test file, maybe `vfs_posix_paths.py`.

Also, we usually have tests in the same commit as the commit that fixes the issue.  So please also try to do that here if it makes sense (I understand you made changes to tests that were accidentally passing, and those can indeed stay in a separate commit.).

Reply:
OK, will do. Probably not before the weekend.

(In case you were wondering about the separate commits, there is a reason I did that. I always do it that way, because it makes it easier to check that 1. something is wrong (by checking out the first commit with the failing test) and 2. the change actually fixes it (by checking out the second commit with the fix). But of course I willl conform to the project style.)

Reply:
I'm fully on board with making lots of small commits and separating out functionality into different commits.  In general we try to do that in this repo.

But for this particular case, I felt that it was a bit confusing which test went with which fix.  And I think it would be easier for someone reading the git history if the fix and test were in the same commit.

Reply:
Done. Apologies it took me several weekends.

Reply:
No problem at all.